projects/plugins/crm/src/automation/class-base-trigger.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Base Trigger implementation
+ *
+ * @package automattic/jetpack-crm
+ */
+
+namespace Automattic\Jetpack\CRM\Automation;
+
+/**
+ * Base Trigger implementation
+ *
+ * @inheritDoc
+ */
+abstract class Base_Trigger implements Trigger {
+
+	/** @var mixed|string The trigger name */
+	protected $name = '';
+	/** @var string The trigger title */
+	protected $title = '';
+	/** @var string The trigger description */
+	protected $description = '';
+	/** @var string The trigger category */
+	protected $category = '';
+
+	/**
+	 * Base_Trigger constructor.
+	 *
+	 * @param array $trigger_data The trigger data.
+	 */
+	public function __construct( array $trigger_data ) {
+		$this->name        = $trigger_data['name'];
+		$this->title       = $trigger_data['title'] ?? '';
+		$this->description = $trigger_data['description'] ?? '';
+		$this->category    = $trigger_data['category'] ?? '';
+	}
```

Comment:
> we don't have scenarios where we don't need any of the data

We could have scenarios where we don't have a category or an empty description, or even a title. For example, internal triggers that we don't want to expose for any reason. The unique required field would be the $name.

Really, since properties are `protected`, the trigger to be implemented, could add their data directly in their constructor without calling the parent constructor:

```
	public function __construct() {
		$this->name 		= 'contact_updated';
		$this->title 		= 'Contact Updated';
		$this->description 	= 'Triggered when a CRM contact is updated';
		$this->category 	= 'contact';
	}
```

In fact, I'd remove the constructor from the `Base_Trigger` since we have to set it from the implemented trigger. The Automation System must be responsible for rejecting/ignoring invalid triggers, for example, without a name.